### PR TITLE
Show a list of possible invoice line type

### LINKED ISSUE #36: Add possible invoice lines types returned
The docs for `/invoices` and `/invoice_lines` endpoints doesn't specify which are the possible values for the `type` returned for invoice lines. 

We need to add a list with this possible values and a short description of each of these types.

### PR DESCRIPTION
**Pull request by cristianbica**
Fixes #36
Preview https://cristian-bica.gitbook.io/test/#line-types

### CODE REVIEW

**Review by dcarneiro — APPROVED** [review]
👌 props for the preview link

**Review by vruizext — COMMENTED** [review]
The current description is providing the same information that one can see reading the line type column. 

AFIU @pootsbook proposed in the support issue that we should provide a description a bit more elaborated:

> Statement about what each type is for/means

https://github.com/socialpaymentsbv/support-issues/issues/2016#issuecomment-621250053

(my bad, I forgot to add a link to the support issue in #36)

I can write down an introductory text and the descriptions. Will post them here later today or next week when I have them ready.

**Comment by cristianbica** [comment]
@vruizext update it with a more detailed explanation (preview here https://cristian-bica.gitbook.io/test/#line-types)

File: api-documentation/invoice-line.md [hunk 1]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `CHARGEBACK-PENALTY-FEE` | | Chargeback for penalty fees |
+
+## Credit Lines
+
+When credits are added to an invocie or fees are credited from the Club Collect Web Interface we will add
+a new invoice line to the invoice.
+
+| Type | Subtype | Description |
+|-|-|-|
+| `CREDIT-LINE` | | General credit line |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: We don't talk of "General" credit lines anywhere. 

I would say, trying to keep the vocabulary / wording consistent:

> Discount applied to the amount invoiced by the partner

File: api-documentation/invoice-line.md [hunk 2]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `CHARGEBACK` | | Chargeback line for regular invoice lines |
+| `CHARGEBACK` | `donation` | Chargeback line for donation invoice lines |
+| `CHARGEBACK-INSTALLMENT-FEE` | | Chargeback for installment fees |
+| `CHARGEBACK-LATE-PAYMENT-FEE` | | Chargeback for late payment fee |
+| `CHARGEBACK-CHARGEBACK-FEE` | | Chargeback for paid chargeback fees |
+| `CHARGEBACK-PENALTY-FEE` | | Chargeback for penalty fees |
+
+## Credit Lines
+
+When credits are added to an invocie or fees are credited from the Club Collect Web Interface we will add
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: I'm not convinced about this wording, it's a bit overcomplicated and giving some details that are unnecessary. Let's keep it simple 

```suggestion
Credit lines are added to the invoice to discount an amount from the amount outstanding. 
```

File: api-documentation/invoice-line.md [hunk 3]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `PAYMENT-INSTALLMENT-FEE` | | Payments for installment fees |
+| `PAYMENT-CHARGEBACK-FEE` | | Payments for chargeback fees |
+| `PAYMENT-LATE-PAYMENT-FEE` | | Payments for late payment fees |
+| `PAYMENT-PENALTY-FEE` | | Payment for penalty fees (Legacy) |
+
+
+## Chargeback Lines
+
+When a previously successful payment is charged back (usually happens for Sepa Direct Direct or Card payments)
+Club Collect will generate chargeback lines for the registered payment lines.
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: I would reword this:

> When a successful payment (usually Sepa Direct Debit) is charged back, chargeback lines are added to the invoice to cancel the amount of the payment that has been charged back.

File: api-documentation/invoice-line.md [hunk 4]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+
+| Type | Subtype | Description |
+|-|-|-|
+| `INSTALLMENT-FEE` | | Fee members must pay when paying for an invoice in installments |
+| `CHARGEBACK-FEE` | | When a payment is charged back a fee will be added to the invoice |
+| `LATE-PAYMENT-FEE` | | Late payment fee added when reminders are sent |
+
+### Payment Lines
+
+When payments are made (either directly by member or external payments added from the Club Collect web interface)
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: I would leave out any mention to "ClubCollect web interface" or to internal logic, trying to keep it simple and straight to the point

> Payment lines are added to register new payments happening for an invoice

File: api-documentation/invoice-line.md [hunk 5]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+payment invoice lines will are added to the invoice.
+
+
+| Type | Subtype | Description |
+|-|-|-|
+| `PAYMENT` | | Payments towards the regular invoice lines |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: > Payment for the amount invoiced by the partner

File: api-documentation/invoice-line.md [hunk 6]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+
+
+## Chargeback Lines
+
+When a previously successful payment is charged back (usually happens for Sepa Direct Direct or Card payments)
+Club Collect will generate chargeback lines for the registered payment lines.
+
+| Type | Subtype | Description |
+|-|-|-|
+| `CHARGEBACK` | | Chargeback line for regular invoice lines |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: > Chargeback for the amount invoiced by the partner

File: api-documentation/invoice-line.md [hunk 7]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+
+
+### Regular Lines
+
+This line type is used for lines added to the invoice at creation time, when increasing the invoice amount
+from the ClubCollect web interface or when a member adds a donation to the invoice (if enabled).
+
+| Type | Subtype | Description |
+|-|-|-|
+| `INVOICE-LINE` | | Regular Invoice line |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: Partners don't know what "regular" invoice lines are. Let's try to explain it in their language (and in the same line as in other pages of the API docs)

> Amounts added by the partner when an invoice is created will be of this type

File: api-documentation/invoice-line.md [hunk 8]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `INVOICE-LINE` | `donation` | Donation Invoice line |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: > Donation added to an invoice

File: api-documentation/invoice-line.md [hunk 9]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `INVOICE-LINE` | `surcharge` | Payment process fee Invoice line |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: > Payment fee charged to the receiver of the invoice

File: api-documentation/invoice-line.md [hunk 10]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+
+### Fees Lines
+
+At certain points of the life cycle of an invoice fees can be added to the invoice (e.g. Member initiating a payment
+in installments).
+
+
+| Type | Subtype | Description |
+|-|-|-|
+| `INSTALLMENT-FEE` | | Fee members must pay when paying for an invoice in installments |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: > These fees are added to the invoice when it's paid in installments

File: api-documentation/invoice-line.md [hunk 11]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `CHARGEBACK-FEE` | | When a payment is charged back a fee will be added to the invoice |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: > When a payment is charged back a fee might be added to the invoice 

(CBFs aren't always added to the invoice, some clubs opt-out to not charge them, others set a threshold of 1 or 2 chargebacks

File: api-documentation/invoice-line.md [hunk 12]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `LATE-PAYMENT-FEE` | | Late payment fee added when reminders are sent |
```
Comment (vruizext) [inline api-documentation/invoice-line.md]: > Late payment fees are penalties that might be added when the invoice is overdue

File: api-documentation/invoice-line.md [hunk 13]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+| `CREDIT-LINE` | | General credit line |
+| `CREDIT-LINE` | `donation` | Credit for donation lines |
+| `CREDIT-LINE` | `surcharge` | Credits for the payment fee |
+| `CREDIT-CHARGEBACK-FEE` | | Credits for a chargeback fee |
+| `CREDIT-INSTALLMENT-FEE` | | Credit for an installment fee |
+| `CREDIT-LATE-PAYMENT-FEE` | | Credit for a late payment fee |
+| `CREDIT-PENALTY-FEE` | | Credit for a Penalty fee |
+
+
+Note: penalty lines are a legacy types. You'll see this in the API for older invoices but no new invoice lines
```
Comment (vruizext) [inline api-documentation/invoice-line.md]:
```suggestion
Note: penalty lines are a legacy type. You might see some of them for older invoices but no new invoice lines of this type are created.
```

File: api-documentation/invoice-line.md [hunk 14]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
 * `paypal`
 * `external`
 {% endhint %}
+
+## Line types
+
+The invoice lines can be of 2 basic types: increasing the invoice balance (regular lines, fees, chargebacks) and will have a positive amount attribute or decreasing the invoice balance (payments, credits) and will have a negative amount attribute. Furthermore lines can be categorized by their type (e.g.: Installment fees, Late payment fees).
```
Comment (vruizext) [inline api-documentation/invoice-line.md]:
```suggestion
The invoice lines can either increase the invoice outstanding (positive amount), e.g. regular lines, fees, chargebacks; or decrease the outstanding (negative amount), e.g. payments, credits. Furthermore lines can be categorised by their type.
```

File: api-documentation/invoice-line.md [hunk 15]
```diff
@@ -161,3 +161,83 @@ Invoice not found.
...
+
+The type of an invoice line is determined by the `type` attributes (always present) and the optional `subtype` attributes. Below is a list of possible values.
```
Comment (vruizext) [inline api-documentation/invoice-line.md]:
```suggestion
The type of an invoice line is determined by the `type` attributes (always present) and the optional `subtype` attributes. Below you can find a description of the possible values for invoice lines types.
```

File: api-documentation/invoice-line.md (hunk 14 above)
Comment (pootsbook) [reply api-documentation/invoice-line.md]: @vruizext `s/whether/either/`

File: api-documentation/invoice-line.md (hunk 10 above)
Comment (pootsbook) [reply api-documentation/invoice-line.md]: Prefer `it is` vs. `it’s`.

**Review by vruizext — COMMENTED** [review]
Looks good in general, left some suggestions to reword some sentence, to make it more in line with the language style we use in API docs

File: api-documentation/invoice-line.md (hunk 6 above)
Comment (pootsbook) [reply api-documentation/invoice-line.md]: Technically the amount is invoiced by the partner. “Legally” the amount is invoiced by the club (via the partner). Not sure if having “by the partner” will clarify or cloud the meaning. The alternatives:

- nothing (`Chargeback`)
- explicit (`Chargeback for the amount invoiced by the partner on behalf of the club`)
- current suggestion (`Chargeback for the amount invoiced by the partner`)

Another technical note, is this not the chargeback of a _payment_ for the amount invoiced, rather than a chargeback for the _amount invoiced_. ?

File: api-documentation/invoice-line.md (hunk 13 above)
Comment (pootsbook) [reply api-documentation/invoice-line.md]: `s/are a legacy types./are legacy types./` OR
`s/are a legacy types./are a legacy type./`

**Review by pootsbook — COMMENTED** [review]
Have given a thumbs up to Victor’s comments where I agree with the change in wording. I have commented in places with additional changes to either Victor’s comments or the original.

File: api-documentation/invoice-line.md (hunk 7 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `INVOICE-LINE` | | Amounts added by the partner when an invoice is created will be of this type |
```

File: api-documentation/invoice-line.md (hunk 8 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `INVOICE-LINE` | `donation` | Donation added to an invoice |
```

File: api-documentation/invoice-line.md (hunk 9 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `INVOICE-LINE` | `surcharge` | Payment fee charged to the receiver of the invoice |
```

File: api-documentation/invoice-line.md (hunk 10 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `INSTALLMENT-FEE` | | These fees are added to the invoice when it is paid in installments |
```

File: api-documentation/invoice-line.md (hunk 11 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `CHARGEBACK-FEE` | | When a payment is charged back a fee might be added to the invoice |
```

File: api-documentation/invoice-line.md (hunk 12 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `LATE-PAYMENT-FEE` | | Late payment fees are penalties that might be added when the invoice is overdue |
```

File: api-documentation/invoice-line.md (hunk 4 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
Payment lines are added to register new payments happening for an invoice.
```

File: api-documentation/invoice-line.md (hunk 5 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `PAYMENT` | | Payment for the amount invoiced by the partner |
```

File: api-documentation/invoice-line.md (hunk 3 above)
Comment (cristianbica) [inline api-documentation/invoice-line.md]:
```suggestion
When a successful payment (usually Sepa Direct Debit) is charged back, chargeback lines are added to the invoice to cancel the amount of the payment that has been charged back.
```

File: api-documentation/invoice-line.md (hunk 1 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `CREDIT-LINE` | | Discount applied to the amount invoiced by the partner |
```

File: api-documentation/invoice-line.md (hunk 6 above)
Comment (cristianbica) [reply api-documentation/invoice-line.md]:
```suggestion
| `CHARGEBACK` | | Chargeback for the amount invoiced by the partner |
```

File: api-documentation/invoice-line.md [hunk 16]
```diff
@@ -176,9 +176,9 @@ from the ClubCollect web interface or when a member adds a donation to the invoi
 
 | Type | Subtype | Description |
 |-|-|-|
-| `INVOICE-LINE` | | Regular Invoice line |
-| `INVOICE-LINE` | `donation` | Donation Invoice line |
-| `INVOICE-LINE` | `surcharge` | Payment process fee Invoice line |
+| `INVOICE-LINE` | | Amounts added by the partner when an invoice is created will be of this type |
```
Comment (vruizext) [inline api-documentation/invoice-line.md:179]: @cristianbica there is an extra ` | ` here?

Comment (vruizext) [reply api-documentation/invoice-line.md:179]: nvm, it's an empty cell

File: api-documentation/invoice-line.md (hunk 6 above)
Comment (vruizext) [reply api-documentation/invoice-line.md]: @pootsbook I used "by the partner" because we use it in other pages in the API docs, IIRC for invoices, we say that "the partner" does this or that.
